TileStache/Goodies/VecTiles/transform.py

```diff
@@ -366,3 +366,90 @@ def tags_name_i18n(shape, properties, fid, zoom):
             properties[alt_tag_name_candidate] = alt_tag_name_value
 
     return shape, properties, fid
+
+# intercut takes features from a base layer and cuts each
+# of them against a cutting layer, splitting any base
+# feature which intersects into separate inside and outside
+# parts.
+#
+# the parts of each base feature which are outside any
+# cutting feature are left unchanged. the parts which are
+# inside have their property with the key given by the
+# 'target_attribute' parameter set to the same value as the
+# property from the cutting feature with the key given by
+# the 'attribute' parameter.
+#
+# the intended use of this is to project attributes from one
+# layer to another so that they can be styled appropriately.
+#
+# returns a set of feature layers with the base layer
+# replaced by a cut one, or None if there's an error.
+def intercut(feature_layers, base_layer, cutting_layer, attribute=None, target_attribute=None):
+    base = None
+    cutting = None
+
+    # the target attribute can default to the attribute if
+    # they are distinct. but often they aren't, and that's
+    # why target_attribute is a separate parameter.
+    if target_attribute is None:
+        target_attribute = attribute
+
+    # search through all the layers and extract the ones
+    # which have the names of the base and cutting layer.
+    # it would seem to be better to use a dict() for
+    # layers, and this will give odd results if names are
+    # allowed to be duplicated.
```

Comment:
Yea, I think it might end up being an invasive change. I haven't double checked, but my gut tells me that the relevant tilestache bits that expect a certain shape from `feature_layers` are the formatters. We can either update those to expect a dict, or just transform the dict into a list right before passing to the formatters, which should be the last step in the chain.

I don't expect anything depends on the ordering currently, although maybe it would make sense to think about that. I would guess that it would match the configuration ordering of the layers in the all layer, but I'm not 100% sure that's the case.

I just created a couple of other issues related to formatters that I had in mind, namely https://github.com/mapzen/tilequeue/issues/23 and https://github.com/mapzen/tilequeue/issues/24, and it might be easier to look at all these more closely together because they touch the same parts of the code.

But yea, I was thinking of looking at all this in the future and not in this pr.